### PR TITLE
Better stun baton script. Remove refs to stun baton in MeleeStun.cs.

### PR DESCRIPTION
**Pull request by PetMudstone**
### Purpose
- Adds a new stun baton script, hopefully cleaner than the old one.
- Removes references to stun baton script and batons in general in MeleeStun.cs
- You no longer hurt people when stunning them on disarm and grab intents with batons and other stunning melee weapons.

Credit to Gilles for helping me with StunBatonV2.cs.

### Changelog:
CL: Removed ability for stun batons to hurt people when used to stun on disarm and grab intent.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using AddressableReferences;
+using Mirror;
+using UnityEngine;
+
+[RequireComponent(typeof(Pickupable))]
+public class StunBatonV2 : NetworkBehaviour, ICheckedInteractable<HandActivate>, IServerSpawn
+{
+	public SpriteHandler spriteHandler;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: no need for this to be public. Make it private and get the component from the children.

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    public MeleeStun meleeStun;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: same here. Make it private and get the components in an Awake method

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        meleeStun = GetComponent<MeleeStun>();
+    }
+
+	// Calls TurnOff() when baton is spawned, see below.
+	public void OnSpawnServer(SpawnInfo info)
+	{
+		TurnOff();
+	}
+
+	// Enables melee stun component and sets baton state to on.
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: no need to over explain the code. This code is perfectly legible, I read a method called "TurnOn" inside the "StunBaton" class, I can already imagine what it does. If I need more details, I read the method body.

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs (hunk 2 above)
Comment (PetMudstone) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: Fixed in fe4b08d. Set to private.

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs (hunk 1 above)
Comment (PetMudstone) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: Fixed in fe4b08d. Set to private.

File: UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs (hunk 3 above)
Comment (PetMudstone) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: Fixed in fe4b08d. Removed a few unnecessary comments.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Weapons/Melee/StunBatonV2.cs]: I'd remove all comments, tbh. That's fine tho, I'm merging.

**Pull request merged**